### PR TITLE
Feat/guppy api tests

### PR DESCRIPTION
**Pull request by ZakirG**

### CODE REVIEW

File: suites/apis/guppyTest.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+const nodes = require('../../utils/nodes.js');
+const fetch = require('node-fetch');
+const fs = require('fs');
+
+Before( async (home, etl, nodes) => {
+  home.complete.login();
+});
+
+// This test fails due to what looks to me like a bug in Guppy that should be fixed. I made a ticket for it.
+xScenario('I want to make a query to Guppy, but my access token is invalid or expired.', async (I, guppy) => {
```
Comment (mfshao) [inline suites/apis/guppyTest.js]:
```suggestion
Scenario('I want to make a query to Guppy, but my access token is invalid or expired.', async (I, guppy) => {
```

File: suites/apis/guppyTest.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+const expect = chai.expect;
+const apiUtil = require('../../utils/apiUtil.js');
+const guppy = require('../../services/apis/guppy/guppyService.js');
+const sheepdog = require('../../services/apis/sheepdog/sheepdogService.js');
+const nodes = require('../../utils/nodes.js');
+const fetch = require('node-fetch');
+const fs = require('fs');
+
+Before( async (home, etl, nodes) => {
+  home.complete.login();
```
Comment (mfshao) [inline suites/apis/guppyTest.js]: for API test you don't need to do this, you can get access token from a user, see comments below

File: suites/apis/guppyTest.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+// This test fails due to what looks to me like a bug in Guppy that should be fixed. I made a ticket for it.
+xScenario('I want to make a query to Guppy, but my access token is invalid or expired.', async (I, guppy) => {
+  let invalidToken = 'eyJhbGciOiJSUzI1N';
+  const queryFile = 'test_plans/guppy/test_data/test_query_1.json';
+  const queryResponse = await guppy.do.submitQueryFileToGuppy(guppy.props.endpoints.graphqlEndpoint, queryFile, invalidToken);
+  expect(queryResponse.status).to.equal(401);
+});
+
+Scenario('I want a list of patients (cases) strictly younger than 30 with a past stroke in ascending order of BMI.', async (I, guppy) => {
+  let token = await I.grabCookie('access_token');
```
Comment (mfshao) [inline suites/apis/guppyTest.js]: you could do something similar to this https://github.com/uc-cdis/gen3-qa/blob/master/suites/apis/userTokenTest.js#L24 to get the access token

File: services/apis/guppy/guppyTasks.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+const guppyProps = require('./guppyProps.js');
+const user = require('../../../utils/user.js');
+const { Gen3Response, getCookie, getAccessTokenHeader } = require('../../../utils/apiUtil');
+const { Bash, takeLastLine } = require('../../../utils/bash');
+const { container } = require('codeceptjs');
+const bash = new Bash();
```
Comment (mfshao) [inline services/apis/guppy/guppyTasks.js]: some of these imports seems never been used

File: suites/apis/guppyTest.js (hunk 1 above)
Comment (ZakirG) [reply suites/apis/guppyTest.js]: the x means "skip". it's a codeceptjs syntax https://codecept.io/basics#skipping

File: services/apis/guppy/guppyTasks.js (hunk 4 above)
Comment (ZakirG) [reply services/apis/guppy/guppyTasks.js]: ur right

File: suites/apis/guppyTest.js (hunk 3 above)
Comment (ZakirG) [reply suites/apis/guppyTest.js]: done

File: services/apis/guppy/guppySequences.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+ */
+module.exports = {
+  async checkQueryResponseEquals(endpoint, queryToSubmitFilename, expectedResponseFilename, accessToken, queryType) {
+    const queryResponse = await guppyTasks.submitQueryFileToGuppy(endpoint, queryToSubmitFilename, accessToken);
+
+    expect(queryResponse.status).to.equal(200);
+
+    let actualResponseJSON = await queryResponse.json();
+    let expectedResponse = fs.readFileSync(expectedResponseFilename).toString();
+    if(queryType == 'aggregation') {
```
Comment (mfshao) [inline services/apis/guppy/guppySequences.js]: probably will be better looking if convert this `if-else` chain into `switch`

**Pull request merged**